jaypeasee/overlook-hotel
# Feature: Manager View

# Pull Request Template

#### What’s this PR do?
In short, this PR creates a passive manager experience. Specifically it:
1. Shows specific metrics for the manager, such as:
* Greets the manager.
* Displays today's date.
* Displays amount of available rooms.
* Displays occupancy percentage.
* Displays the total revenue for today.
2. Shows forms to:
* Look up a guest.
* Search for room availability by date.
3. A guest view lookup that shows that guests past, present and future bookings
4. A date view lookup that shows availability by date and the ability to book for a specific guest.
5. Refactors the `base.scss` stylesheet to make more use of `mixins` and a `mixins.scss` file.

###### Side Note:
According to the `Lighthouse` review, the accessibility rating is still at 100%.

#### Where should the reviewer start?
* Start on line 185 of `index.js` and the functions it invokes from there.
* Review `manager.js` and `manager-test.js` to see how those class methods changed.
* Review `base.scss` to see how the html inserted via js is styled.


#### How should this be manually tested?
1. Log in to the manager view with these credentials:
* Username: `manager`
* Password: `overlook2020`
1. View the nav to make sure all metrics are accurate.
1. Enter a valid and invalid customer's name in the customer lookup form to see the results.
* When a valid customer is entered, you should see that customer's booking history (past, present, and future).
1. Enter a valid and invalid date in the date lookup to see the results.
* When a valid date is entered, you should see an input for a name and a button to book for a customer.

#### Any background context you want to provide?
The next PR will include POST requests for when the customer and manager book rooms.

#### What are the relevant tickets?
* #28 
* #29 
* #30 
* #34 

#### Screenshots (if appropriate):
![Screen Shot 2020-11-07 at 10 34 06 AM](https://user-images.githubusercontent.com/59029768/98447809-00ec0700-20e5-11eb-8dc7-2b1e01cb682a.png)

![Screen Shot 2020-11-07 at 10 35 30 AM](https://user-images.githubusercontent.com/59029768/98447810-02b5ca80-20e5-11eb-9351-a41b6b4ec6e5.png)
